unified-runtime/source/adapters/level_zero/image_common.cpp

```diff
@@ -274,7 +274,15 @@ ur_result_t createUrImgFromZeImage(ze_context_handle_t hContext,
                                    ur_exp_image_mem_native_handle_t *pImg) {
   ze_image_handle_t ZeImage;
   ZE2UR_CALL(zeImageCreate, (hContext, hDevice, &ZeImageDesc, &ZeImage));
-  ZE2UR_CALL(zeContextMakeImageResident, (hContext, hDevice, ZeImage));
+  if (ZeImage == nullptr) {
+    return UR_RESULT_ERROR_OUT_OF_HOST_MEMORY;
+  }
+  try {
+    ZE2UR_CALL_THROWS(zeContextMakeImageResident, (hContext, hDevice, ZeImage));
+  } catch (const ze_result_t &result) {
+    ZE2UR_CALL(zeImageDestroy, (ZeImage));
```

Comment:
Instead of manually calling zeImageDestory (which is error prone) it would be better to use v2::raii:ze_image_handle_t